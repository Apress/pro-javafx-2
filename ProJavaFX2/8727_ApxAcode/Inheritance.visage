/*
 * @test
 * @run
 */

class A {
  var b: Boolean;
  function f() {
    println("A.f() called: b is {b}.");
  }
}

mixin class B {
  var i: Integer;
  function g() {
    println("B.g() called: i is {i}");
  }
}

class C extends A, B {
  var n: Number;
  function h() {
    println("C.h() called: b = {b}, i = {i}, n = {n}.");
  }
}

var c = C { b: true, i: 1024, n: 3.14 };
c.f();
c.g();
c.h();
