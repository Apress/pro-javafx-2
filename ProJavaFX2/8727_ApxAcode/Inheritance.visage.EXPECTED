A.f() called: b is true.
B.g() called: i is 1024
C.h() called: b = true, i = 1024, n = 3.14.
